src/clover/fileparser.py

```diff
@@ -2176,6 +2176,79 @@ def _parse_transmission_inputs(
         transmitters,
     )
 
+# Grid inputs file:
+#   Keyword used for parsing diesel-generator information.
+GRID: str = "grid"
+GRID_EMISSIONS: str="emissions"
+GridTier: str="tier"
+GridType: str="type"
+# # The relative path to the grid inputs file.
+
+GRID_INPUTS_FILE: str = os.path.join("generation", "grid_inputs.yaml")
+
+def _parse_grid_inputs(
+    inputs_directory_relative_path: str,
+    logger: Logger,
+    scenarios: List[Scenario],
+) -> List[GRID]:
+
+    grid_inputs_filepath = os.path.join(
+        inputs_directory_relative_path, GRID_INPUTS_FILE
+    )
+    grid_inputs = read_yaml(grid_inputs_filepath, logger)
+    if not isinstance(grid_inputs, dict):
+        raise InputFileError(
+            "Grid inputs", "Grid input file is not of type `list`."
+        )
+    logger.info("Grid inputs successfully parsed.")
+
+    grids: List[Grid] = []
+    for entry in grid_inputs:
+        tiers: List[GridTier] = []
```

Comment:
So, here, while you __think__ you're creating a list of `GridTier` instances, you're actually making a `list` of `strings`, because `GridTier` doesn't point to the class we defined in the utility module, but to the string we defined just above

Reply:
But in my class it is written as GridTier ... should it become here tiers: List[GRID_TIER]?

Reply:
No, this is right :smile: but you've overwritten `GridTier` in the above comment (line 2183) which means that this line, while fine, won't work until you fix line 2183 :smile:

Reply:
@BenWinchester